client/src/routes/automation/esidom_generator.ts

```diff
@@ -0,0 +1,256 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable camelcase */
+import Blockly from 'blockly';
+import type { Block } from 'blockly';
+
+const esidomGenerator = new Blockly.Generator('ESIDOM');
+
+const PRECEDENCE = 0;
+
+interface BlocklyJSON {
+    trigger?: string;
+    condition?: string;
+    action?: string;
+    platform?: string;
+    at?: string;
+    service?: string;
+    entity_id?: string;
+    after?: string;
+    before?: string;
+    weekday?: string[];
+    from?: string;
+    to?: string;
+    state?: string;
+    rgb_color?: string;
+}
+
+interface BlocksGenerator {
+    automation: (blk: Block) => void;
+    binary_trigger: (blk: Block) => void;
+    time: (blk: Block) => void;
+    time_condition: (blk: Block) => void;
+    sun_condition: (blk: Block) => void;
+    time_condition_hour: (blk: Block) => void;
+    time_condition_week: (blk: Block) => void;
+    binary_condition: (blk: Block) => void;
+    action: (blk: Block) => void;
+    color_picker: (blk: Block) => void;
+    color_rgb: (blk: Block) => void;
+    scrub: (blk: Block, code: string, opt_thisOnly: string) => void
+    jsonInit:(a: Block)=> void;
+}
+
+((block: BlocksGenerator) => {
+    block.automation = (blk: Block) => {
+        const statements_trigger = esidomGenerator.statementToCode(blk, 'Trigger');
+        const statements_condition = esidomGenerator.statementToCode(blk, 'Condition');
+        const statements_action = esidomGenerator.statementToCode(blk, 'Action');
+        const dropdown_mode = blk.getFieldValue('Mode');
```

Comment:
Does this variable will be use later ? Currently it seems useless

Reply:
It will be removed with the new "automation" block definition coming from Anais.

Reply:
Just keep in mind to send this value to the middle/HA. It could be used later. Actually, it could be a default value of the block even if it doesn't come from user